lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,54 @@
+class LRUCache
+  extend Forwardable
+  def_delegators :@table, :size, :count
+  def_delegators :@table, :empty?
+  def_delegators :@table, :keys
+  def_delegators :@table, :clear
+
+  attr_accessor :table, :minimum_time_in_queue
+  def initialize(minimum_time_in_queue = 5)
+    @minimum_time_in_queue = minimum_time_in_queue * 60
+    @table = {}
+  end
+
+  def set(key, value)
+    @table.delete(key)
+    @table[key] = value
+    clear_unused_resources
+    value
+  end
+
+  def get(key)
+    if found = @table.delete(key)
+      @table[key] = found
+    end
+  end
+
+  def []=(key, value)
+    set(key, value)
+  end
+
+  def [](key)
+    get(key)
+  end
+
+  def delete(key)
+    @table.delete(key)
+  end
+
+  private
+
+  def clear_unused_resources
+    cleaned_elements = 0
+    @table.each do |key, value|
+      break if cleaned_elements >= 2
+      next if value.instance_of?(Semian::Resource)
```

Comment:
For some reason, Github is not letting me comment where you addressed this. I don't see why this would ever be true. If a test is violating this, let's make sure that test is using a protected resource IMO.